editor/src/components/text-editor/text-editor-shortcut-helpers.ts

```diff
@@ -68,3 +190,36 @@ const toggleStyleProp = (
     jsxAttributeValue(newValue, emptyComments),
   )
 }
+
+const toggleStylePropWithUnset = (
+  elementPath: ElementPath,
+  prop: string,
+  currentValue: string | null,
+  toggledValue: string,
+  defaultValue: string,
+  dispatch: EditorDispatch,
+  getComputedValue: () => string | null,
+): void => {
+  const newValue = currentValue === toggledValue ? defaultValue : toggledValue
+  if (newValue === defaultValue) {
+    dispatch([EditorActions.unsetProperty(elementPath, PP.create('style', prop))])
+    const computedValue = getComputedValue()
+    if (computedValue !== newValue) {
+      dispatch([
+        EditorActions.setProperty(
+          elementPath,
+          PP.create('style', prop),
+          jsxAttributeValue(newValue, emptyComments),
+        ),
+      ])
+    }
```

Comment:
can these two dispatches be combined into a single one? will this cause double undo?

Reply:
they cannot be combined, however, the second action dispatched should be wrapped in a `transientActions()` – until we have the proposed "appendToPreviousAction" behavior

Reply:
done